### ISSUE TITLE
`.toggle` is hard to understand and should be DEPRECATED

### ISSUE THREAD

**Issue by lizmat** [Raku/problem-solving#450]
See discussion at https://github.com/rakudo/rakudo/issues/2089 and the issue https://github.com/rakudo/rakudo/issues/2088.

**Comment by librasteve** [comment]
read some of the references, I agree that `toggle` is too hard and that the main practical concern is the effect of varying data on the consumption of the test chain

I can see in a Method chain, or a Supply / Tap context that it is useful to have functions that:
  a. filter values (eg to cap, collar real world data perhaps)
  b. turn on / off the feed depending on value gates within the data (eg inspect the contents of "packets")

(a) can be done with `grep: -5 < * <= 5` 
(b) can be done with a modification to `toggle`

therefore I propose the following modification / simplification of `toggle`:

*`toggle` should only take one test which should be repeatedly applied to each element in the list (ie the notion of stepping along the args should be scrapped)*

`(1,2,3,0,3,3,3,0,1,2,0,2,2,0).toggle( *==0, :off )  => 3,3,3,2,2`

Callable & Matcher both seem good for this use case

Apologies if there is another good way to do this in raku that I do not know

Another adverb pair `:include`, `:exclude` (default include) is needed to gate whether the tested element is passed through

EDIT: based on @lizmat example below, I will withdraw this idea

**Comment by lizmat** [comment]
```
% raku -e 'my $on; say (1,2,3,0,3,3,3,0,1,2,0,2,2,0).map({ if $_ == 0 { $on = !$on; Empty } elsif $on { $_ } })'
(3 3 3 2 2)
```

**Comment by CIAvash** [comment]
```raku
say (1,2,3,0,3,3,3,0,1,2,0,2,2,0).grep({ 0 ^fff^ 0 })
=output (3 3 3 2 2)
```

**Comment by 2colours** [comment]
I tend to agree that `.toggle` never really caught on and the abstraction it provides is neither efficient nor particularly useful. However, I also agree with @FCO who [went ahead and created this module](https://github.com/FCO/SeqSplitter) that "takewhile", "dropwhile" kind of functionality _is_ useful and currently the only reasonable functional (non-gather-take) core way to achieve it seems to be `.toggle`. This is something to consider for the future.
